src/app/banners/BannerManager.ts

```diff
@@ -8,35 +8,39 @@ const BANNER_NAME_FOR_FREE_USERS = 'show_free_users_banner';
 export class BannerManager {
   private readonly plan: PlanState;
   private readonly offerEndDay: Date;
-  private readonly bannerItemInLocalStorage: string | null;
   private readonly todayDate: string;
 
+  private readonly localStorageKeys = {
+    freeUsers: BANNER_NAME_FOR_FREE_USERS,
+    general: BANNER_NAME_IN_LOCAL_STORAGE,
+  };
+
   constructor(user: UserSettings, plan: PlanState, offerEndDay: Date) {
     this.plan = plan;
     this.offerEndDay = offerEndDay;
-    this.bannerItemInLocalStorage = localStorageService.get(BANNER_NAME_IN_LOCAL_STORAGE);
     this.todayDate = new Date().toISOString().split('T')[0];
   }
 
   private isOfferExpired(): boolean {
     return new Date() > this.offerEndDay;
   }
 
-  private isLocalStorageExpired(): boolean {
-    return (this.bannerItemInLocalStorage ?? '') < this.todayDate;
+  private isLocalStorageExpired(key: string): boolean {
+    const storedDate = localStorageService.get(key) ?? '';
+    return storedDate < this.todayDate;
```

Comment:
This function is not right, you can pass it any key and it doesn't have to return a date. Check it and add tests for the `BannerManager`